File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/Classes/MultipleSubclassesWithOverrides.swift
@main
struct MyMain {

    static func main() {
        let someVehicle = Vehicle()
        let train = Train()
        train.makeNoise()
        let car = Car()
        car.currentSpeed = 25.0
        car.gear = 3
        let carDes = car.description
    }
}
class Vehicle {
    var currentSpeed = 0.0
    var description: String {
        return "traveling at \(currentSpeed) miles per hour"
    }
    func makeNoise() {
        // do nothing - an arbitrary vehicle doesn't necessarily make a noise
    }
}

class Train: Vehicle {
    override func makeNoise() {
        print("Choo Choo")
    }
}

//If you provide a setter as part of a property override, you must also
//provide a getter for that override. If you don’t want to modify
//the inherited property’s value within the overriding getter, you
//can simply pass through the inherited value by returning
//super.someProperty from the getter, where someProperty is the name
//of the property you are overriding.
class Car: Vehicle {
    var gear = 1
    override var description: String {
        return super.description + " in gear \(gear)"
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-1/MultipleSubclassesWithOverrides.cpp

 #include <iostream>

class Vehicle 
{
  public:
    double currentSpeed = 0.0;
    std::string description() {
        return "traveling at "+std::to_string( currentSpeed )+" miles per hour";
    }
    
    void makeNoise() {
    }
};

class Train : public Vehicle 
{
public:
    void makeNoise() {
        std::cout << "Choo Choo";
    }
};

class Car : public Vehicle 
{
public:
    int gear = 1;
    std::string description() {
        return Vehicle::description() + " in gear "+std::to_string( gear );
    }
};

int main()  
{
    Vehicle someVehicle = Vehicle();
    Train train = Train();
    train.makeNoise();
    Car car = Car();
    car.currentSpeed = 25.0;
    car.gear = 3;
    std::string carDes = car.description();
    return 0;
}